drivers/spi/spi_silabs_eusart.c

```diff
@@ -236,6 +241,61 @@ static int spi_silabs_eusart_configure(const struct device *dev, const struct sp
 	return 0;
 }
 
+static inline void spi_silabs_eusart_pm_policy_get(const struct device *dev)
+{
+	pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+	pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
+}
+
+static inline void spi_silabs_eusart_pm_policy_put(const struct device *dev)
+{
+	pm_policy_state_lock_put(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+	pm_policy_state_lock_put(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
+}
+
+static int spi_silabs_eusart_pm_action(const struct device *dev,
+						     enum pm_device_action action)
+{
+	const struct spi_silabs_eusart_config *eusart_config = dev->config;
+	int ret;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		ret = clock_control_on(eusart_config->clock_dev,
+				       (clock_control_subsys_t)&eusart_config->clock_cfg);
+
+		if (ret == -EALREADY) {
+			/* Ignore if clock is already on */
+			ret = 0U;
+		} else if (ret < 0) {
+			return ret;
+		}
+
+		pinctrl_apply_state(eusart_config->pcfg, PINCTRL_STATE_DEFAULT);
+
+		break;
```

Comment:
`EUSART_Enable(eusart_config->base, eusartEnable)` during the resume. Is it expected?

Reply:
So, the call to `clock_control_on()` is also useless?

Reply:
Maybe you could add this comment in the code.